platform/macos/src/MGLMapView.h

```diff
@@ -403,6 +404,19 @@ MGL_EXPORT IB_DESIGNABLE
 - (void)setVisibleCoordinateBounds:(MGLCoordinateBounds)bounds edgePadding:(NSEdgeInsets)insets animated:(BOOL)animated;
 
 /**
+ Changes the receiver’s viewport to fit the given shape, with the specified direction,
+ and optionally some additional padding on each side.
+
+ @param shape The shape that the viewport will show in its entirety.
+ @param direction The direction of the viewport, measured in degrees clockwise from true north.
+ @param insets The minimum padding (in screen points) that will be visible
+    around the given coordinate bounds.
+ @param animated Specify `YES` to animate the change by smoothly scrolling and
+    zooming or `NO` to immediately display the given bounds.
+ */
+- (void)setVisibleShape:(MGLShape *)shape direction:(double)direction edgePadding:(NSEdgeInsets)insets animated:(BOOL)animated;
```

Comment:
I continue to think this would be a poor API to expose publicly, per https://github.com/mapbox/mapbox-gl-native/pull/10107#discussion_r142760198. For one thing, it isn’t possible to customize the duration or timing function or provide a completion handler, all options that come with `-setCamera:withDuration:animationTimingFunction:edgePadding:completionHandler:` on iOS.

If it’s really necessary to achieve a padding-less camera fitting a shape, can we remove this method, replace it with `-setCamera:withDuration:animationTimingFunction:edgePadding:completionHandler:` from iOS, and use the following code to achieve the same effect for the time being?

```objc
MGLMapCamera *camera = [mapView cameraThatFitsShape:shape direction:direction edgePadding:edgePadding];
NSEdgeInsets edgePadding = contentInsets;
edgePadding.left *= -1; edgePadding.right *= -1; edgePadding.top *= -1; edgePadding.bottom *= -1;
[mapView setCamera:camera withDuration:5 animationTimingFunction:nil edgePadding:edgePadding completionHandler:nil];
```